cirq/google/common_serializers_test.py

```diff
@@ -231,6 +231,63 @@ def test_half_pi_does_not_serialize_arbitrary_xy():
         HALF_PI_GATE_SET.serialize_op_dict(gate(q))
 
 
+@pytest.mark.parametrize(('x_exponent', 'z_exponent', 'axis_phase_exponent'), [
+    (0, 0, 0),
+    (1, 0, 0),
+    (0, 1, 0),
+    (0.5, 0, 0.5),
+    (0.5, 0.5, 0.5),
+    (0.25, 0.375, 0.125),
+])
+def test_serialize_deserialize_arbitrary_xyz(
+        x_exponent,
+        z_exponent,
+        axis_phase_exponent,
+):
+    gateset = cg.serializable_gate_set.SerializableGateSet(
+        gate_set_name='test_xyz',
+        serializers=([cgc.PHASED_X_Z_SERIALIZER]),
+        deserializers=([cgc.PHASED_X_Z_DESERIALIZER]))
+    gate = cirq.PhasedXZGate(
+        x_exponent=x_exponent,
+        z_exponent=z_exponent,
+        axis_phase_exponent=axis_phase_exponent,
+    )
+    op = gate.on(cirq.GridQubit(1, 2))
+    expected = {
+        'gate': {
+            'id': 'xyz'
+        },
+        'args': {
+            'x_exponent': {
+                'arg_value': {
+                    'float_value': x_exponent
+                }
+            },
+            'z_exponent': {
+                'arg_value': {
+                    'float_value': z_exponent
+                }
+            },
+            'axis_phase_exponent': {
+                'arg_value': {
+                    'float_value': axis_phase_exponent
+                }
+            }
+        },
+        'qubits': [{
+            'id': '1_2'
+        }]
+    }
+    assert gateset.serialize_op_dict(op) == expected
+    deserialized_op = gateset.deserialize_op_dict(expected)
+    cirq.testing.assert_allclose_up_to_global_phase(
+        cirq.unitary(deserialized_op),
+        cirq.unitary(op),
+        atol=1e-7,
```

Comment:
All the input values you've chosen can be represented as floats exactly, so I wonder whether we could test this for exact equality?

I'm concerned that we may be missing an issue where a float is serialized as string with insufficient number of characters.

Reply:
Many tests in this module have the same issue, in that we are comparing serialized representations but we have to pick certain values to get exact equality of floats. I agree this is something we should solve, but perhaps we can do that in a separate PR for the entire module, rather than a one-off for this new case. WDYT?

Reply:
Yeah, this is definitely out of the scope of this PR.

FTR from our offline discussion: serialization and deserialization of values such as 0.5 or 0.125 that are represented exactly in both single and double precision floats should yield result exactly equal to the original value. We should not be throwing our hands up just because floats ;-)